lib/util/index.js

```diff
@@ -591,6 +580,44 @@ function createMediaSignalingPayload(dominantSpeaker, networkQuality, trackSwitc
     );
 }
 
+/**
+ * Create {@link VideoTrack.Dimensions} RSP payload.
+ * @param {VideoTrack.Dimensions} [dimensions]
+ * @returns {object}
+ */
+function createDimensionsPayload(dimensions) {
+  return ['height', 'width'].reduce((payload, dimension) => {
+    const value = dimensions[dimension];
+    if (typeof value === 'number') {
+      payload = Object.assign({
+        [dimension]: value
+      }, payload);
+    }
+    return payload;
+  }, {});
+}
+
+/**
+ * Create {@link VideoRenderDimensions} RSP payload.
+ * @param renderDimensions
+ * @returns {object}
+ */
+function createRenderDimensionsPayload(renderDimensions) {
+  return [
+    ['dominantSpeaker', 'active_speaker']
+  ].concat(Object.values(constants.trackPriority).map(priority => {
+    return [priority];
+  })).reduce((payload, [prop, payloadProp = prop]) => {
+    const value = renderDimensions[prop];
+    if (typeof value === 'object') {
+      payload = Object.assign({
```

Comment:
```suggestion
      payload[payloadProp] = createDimensionsPayload(value);
```

Reply:
@makarandp0 Then we would have to add payload.video = payload.video || {} everywhere.